odoo/tools/pdf.py

```diff
@@ -94,7 +96,14 @@ def add_banner(pdf_stream, text=None, logo=False, thickness=2 * cm):
     :return (BytesIO):              The modified PDF stream.
     """
 
-    old_pdf = PdfFileReader(pdf_stream, strict=False, overwriteWarnings=False)
+    try:
+        old_pdf = PdfFileReader(pdf_stream, strict=False, overwriteWarnings=False)
+        old_pdf.getNumPages()
+    except Exception:
+        # Import is done here to prevent circular import issues.
+        from odoo.tools.translate import _
```

Comment:
I think we should avoid those kinds of hacks if possible, it indicates that we're breaking the normal code dependencies somehow 🤨 (business logic call helpers, helpers don't take business logic decisions)

Besides, there are some other pdf helpers that could blow up in a similar fashion.

How about we handle this in a consistent manner for all the helpers?
- we catch all parsing errors, as you did
- we log the root cause in INFO level, for technical diagnostics if needed
- we wrap the original exception in a generic, low-level `ValueError`, raise it and let the caller decide what to do?

It would make sense to let the callers decide if/how they want to deal with a possible parsing error, no? They could provide a more contextual feedback to the user (does the error matter? should it be visible or silenced? etc), and then handle translations too.

Reply:
> I think we should avoid those kinds of hacks if possible, it indicates that we're breaking the normal code dependencies somehow raised_eyebrow (business logic call helpers, helpers don't take business logic decisions)

I think it's mostly because `odoo.tools.__init__` is an ill-ordered mess of eager imports: `from . import pdf` precedes `from .translate import *`, so when `pdf` forces an import of its sibling (from the package root as well because relative imports are for chumps) it changes the entire resolution order.

`from . import pdf` could just be moved to the end of the file if it needs to depend on `translate`. And it should use a relative import.